packages/venia-ui/lib/components/CategoryList/categoryList.js

```diff
@@ -1,96 +1,90 @@
-import React, { Component } from 'react';
+import React from 'react';
 import { string, number, shape } from 'prop-types';
-import { Query } from '@magento/venia-drivers';
-import classify from '../../classify';
+import { mergeClasses } from '../../classify';
 import { fullPageLoadingIndicator } from '../LoadingIndicator';
 import defaultClasses from './categoryList.css';
 import CategoryTile from './categoryTile';
 import categoryListQuery from '../../queries/getCategoryList.graphql';
+import { useCategoryList } from '@magento/peregrine/lib/talons/CategoryList/useCategoryList';
 
-class CategoryList extends Component {
-    static propTypes = {
-        id: number,
-        title: string,
-        classes: shape({
-            root: string,
-            header: string,
-            content: string
-        })
+// map Magento 2.3.1 schema changes to Venia 2.0.0 proptype shape to maintain backwards compatibility
+const mapCategory = categoryItem => {
+    const { items } = categoryItem.productImagePreview;
+    return {
+        ...categoryItem,
+        productImagePreview: {
+            items: items.map(item => {
+                const { small_image } = item;
+                return {
+                    ...item,
+                    small_image:
+                        typeof small_image === 'object'
+                            ? small_image.url
+                            : small_image
+                };
+            })
+        }
     };
+};
 
-    get header() {
-        const { title, classes } = this.props;
+const CategoryList = props => {
+    const { id, title } = props;
+    const mixinProps = useCategoryList({
+        query: categoryListQuery,
+        id
+    });
 
-        return title ? (
-            <div className={classes.header}>
-                <h2 className={classes.title}>
-                    <span>{title}</span>
-                </h2>
-            </div>
-        ) : null;
-    }
+    const { data, error, loading } = mixinProps;
```

Comment:
I could probably make this simpler by just returning the error and the normalized data, `categoryChildren`.